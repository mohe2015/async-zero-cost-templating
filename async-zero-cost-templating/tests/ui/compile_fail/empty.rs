extern crate alloc;

use async_zero_cost_templating::html;
use async_zero_cost_templating::TemplateToStream;
use core::pin::pin;
use futures_util::stream::StreamExt;
use std::borrow::Cow;

#[tokio::test]
async fn test() {
    let stream = html! {};
    let result: String = stream.collect().await;
    assert_eq!(result, r#""#)
}
