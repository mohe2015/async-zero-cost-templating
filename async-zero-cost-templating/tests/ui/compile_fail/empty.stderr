error[E0282]: type annotations needed for `TemplateToStream<_, ...>`
 --> tests/ui/compile_fail/empty.rs:6:9
  |
6 |     let _ = html! { };
  |         ^
  |
help: consider giving this pattern a type, where the placeholders `_` are specified
  |
6 |     let _: TemplateToStream<_, {async block@$DIR/tests/ui/compile_fail/empty.rs:6:13: 6:22}> = html! { };
  |          +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
